Add driver, car, and manufacturer detail and list views

frezworx opened the pull request:
Implemented views and templates for driver, car, and manufacturer details and lists. Enhanced base template with pagination includes and fixed URL links in the sidebar.
![ss_1](https://github.com/user-attachments/assets/4bca2273-b560-44ec-89a4-9b9d02e04601)
![ss_2](https://github.com/user-attachments/assets/49f80930-3874-4996-84de-1f0cb8b14f35)
![ss_3](https://github.com/user-attachments/assets/da94d3ea-2c09-428d-83cd-b3682ea2951c)
![ss_4](https://github.com/user-attachments/assets/3c561cf8-151c-4b57-bb67-cde4969a1931)
![ss_5](https://github.com/user-attachments/assets/e293c61c-1075-437e-9027-7b4aff56f2f1)

frezworx committed:
Add driver, car, and manufacturer detail and list views

Implemented views and templates for driver, car, and manufacturer details and lists. Enhanced base template with pagination includes and fixed URL links in the sidebar.

mateacademy-ai-mentor commented on taxi/urls.py:
```diff
@@ -1,9 +1,23 @@
+    path(
+        "drivers/<int:pk>/", DriverDetailView.as_view(), name="driver-detail"
+    ),
+    path(
+        "manufacturers/", ManufacturerListView.as_view(), name="manufacturer-list"
+    ),
```
There is a duplicate path for 'manufacturers/'. It is already defined in lines 8-11. Consider removing one of these duplicate entries to avoid conflicts.

mateacademy-ai-mentor commented on taxi/views.py:
```diff
@@ -1,4 +1,7 @@
+from symtable import Class
```
It seems like the import `from symtable import Class` is not used in this file. Consider removing it if it's unnecessary.

mateacademy-ai-mentor commented on templates/includes/pagination.html:
```diff
@@ -0,0 +1,27 @@
+  
+    {% if page_obj.has_previous %}
+      <li class="page-item">
+        <a href="?page= {{ page_obj.previous_page_number }}" class="page-link">
```
There is an extra space after the '=' in the href attribute. It should be '?page={{ page_obj.previous_page_number }}' to ensure the URL is correctly formed.

mateacademy-ai-mentor commented on templates/includes/pagination.html:
```diff
@@ -0,0 +1,27 @@
+
+    {% if page_obj.has_next %}
+      <li class="page-item">
+        <a href="?page= {{ page_obj.next_page_number }}" class="page-link">
```
There is an extra space after the '=' in the href attribute. It should be '?page={{ page_obj.next_page_number }}' to ensure the URL is correctly formed.

mateacademy-ai-mentor commented on templates/taxi/car_detail.html:
```diff
@@ -0,0 +1,25 @@
+      <td>{{ car.manufacturer.name }}</td>
+      <td>{{ car.manufacturer.country }}</td>
+      <td>
+        {% for driver in car.drivers.all %}
```
There is a missing pair of parentheses in 'car.drivers.all'. It should be 'car.drivers.all()' to properly call the method and retrieve the list of drivers.

mateacademy-ai-mentor commented on templates/taxi/driver_detail.html:
```diff
@@ -0,0 +1,24 @@
+        </tr>
+        {% empty %}
+        <tr>
+          <td colspan="2">The driver doesn't a car</td>
```
There is a grammatical error in the message. It should be 'The driver doesn't have a car' instead of 'The driver doesn't a car'.

mateacademy-ai-mentor commented on templates/taxi/driver_list.html:
```diff
@@ -0,0 +1,34 @@
+      </tr>
+      {% empty %}
+      <tr>
+        <td colspan="5"></td>
```
Consider adding a message inside the empty cell to inform users that there are no drivers available. This will improve user experience by providing feedback when the list is empty.

mateacademy-ai-mentor reviewed (approved):
Great job on setting up the models and templates! 🎉 The issues mentioned are minor and don't affect the core functionality. Keep up the good work and consider addressing these small improvements in the future. 😊

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

frezworx committed:
Fix pagination and car detail bugs, remove manufacturer URL.

Corrected href spacing in pagination links and fixed method call for car drivers in the car detail template. Removed unused manufacturer URL configuration and cleaned up unnecessary import in views.

frezworx committed:
Fix template typos and improve placeholder texts

Corrected method calling conventions and fixed a typo in driver_detail.html. Updated placeholder text in driver_list.html to provide clearer user feedback.

frezworx committed:
Reformat manufacturer URL path for readability.

Broke the "manufacturers" URL path into multiple lines to enhance readability and maintain code consistency. This change has no impact on functionality but improves code maintainability.

frezworx committed:
Fix car list rendering in driver detail template

Removed unnecessary parentheses in the 'for' loop iteration to correct the car list rendering. This ensures that the template correctly displays the list of cars associated with each driver.
